picoquic/sender.c

```diff
@@ -3860,8 +3860,8 @@ static int picoquic_check_idle_timer(picoquic_cnx_t* cnx, uint64_t* next_wake_ti
             idle_timer = UINT64_MAX;
         }
     }
-    else if (cnx->local_parameters.idle_timeout > (PICOQUIC_MICROSEC_HANDSHAKE_MAX / 1000)) {
-        idle_timer = cnx->start_time + cnx->local_parameters.idle_timeout*1000ull;
+    else if (cnx->local_parameters.max_idle_timeout > (PICOQUIC_MICROSEC_HANDSHAKE_MAX / 1000)) {
+        idle_timer = cnx->start_time + cnx->local_parameters.max_idle_timeout*1000ull;
     }
     else {
         idle_timer = cnx->start_time + PICOQUIC_MICROSEC_HANDSHAKE_MAX;
```

Comment:
This becomes the **connection timeout** before negotiation happens when `max_idle_timeout` is less than `PICOQUIC_MICROSEC_HANDSHAKE_MAX`.  It's currently hard coded to a minimum of 30 seconds. It would be nice if we had a configuration for connection timeout.